bundles/org.openhab.core.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/service/ConfigurableServiceResource.java

```diff
@@ -308,19 +311,56 @@ private String getConfigDescriptionByFactoryPid(String factoryPid) {
     }
 
     private String getServiceId(ServiceReference<?> serviceReference) {
+        final String cn = (String) serviceReference.getProperty(ComponentConstants.COMPONENT_NAME);
         Object pid = serviceReference.getProperty(Constants.SERVICE_PID);
-        if (pid != null) {
-            if (pid instanceof String) {
-                return (String) pid;
-            } else if (pid instanceof List) {
-                @SuppressWarnings("unchecked")
-                List<String> pidList = (List<String>) pid;
-                if (!pidList.isEmpty()) {
-                    return pidList.get(0);
+        if (pid == null) {
+            return cn;
+        }
+
+        final String serviceId;
+        if (pid instanceof String) {
+            serviceId = (String) pid;
+        } else if (pid instanceof String[]) {
+            final String[] pids = (String[]) pid;
+            serviceId = getServicePID(cn, Arrays.asList(pids));
+        } else if (pid instanceof Collection) {
+            Collection<?> pids = (Collection<?>) pid;
+            serviceId = getServicePID(cn, pids.stream().map(entry -> entry.toString()).collect(Collectors.toList()));
+        } else {
+            logger.warn("The component \"{}\" is using an unhandled service PID type ({}). Use component name.", cn,
+                    pid.getClass());
+            serviceId = cn;
```

Comment:
We can return the component name directly and skip the rest.

```suggestion
            return cn;
```